src/Serialization/containers.jl

```diff
@@ -3,77 +3,115 @@
 
 @registerSerializationType(Vector)
 
-@doc raw"""
-    is_basic_serialization_type(::Type)
-
-During the serialization of types of the form `Vector{T}`, entries
-of type `T` will either be serialized as strings if `is_basic_serialization_type`
-returns `true`, or serialized as a dict provided the serialization for such a `T`
-exists. If `Vector{T}` is serialized with `is_basic_serialization_type(T) = true`
-then the `entry_type` keyword is used to store the type `T` as a property of
-the vector.
-
-# Examples
-
-```jldoctest
-julia> is_basic_serialization_type(ZZRingElem)
-true
-```
-"""
-is_basic_serialization_type(::Type) = false
-is_basic_serialization_type(::Type{ZZRingElem}) = true
-is_basic_serialization_type(::Type{QQFieldElem}) = true
-is_basic_serialization_type(::Type{Bool}) = true
-is_basic_serialization_type(::Type{Symbol}) = true
-is_basic_serialization_type(::Type{T}) where T <: Number = isconcretetype(T)
-
 function save_internal(s::SerializerState, vec::Vector{T}) where T
     d = Dict{Symbol, Any}(
         :vector => [save_type_dispatch(s, x) for x in vec]
     )
     if is_basic_serialization_type(T)
         d[:entry_type] = encodeType(T)
     end
+
+    if !(T <: RingElem)
+        return d
+    end
+    entry_parent = parent(vec[1])
+    if has_elem_basic_encoding(entry_parent)
+        d[:parent] = save_type_dispatch(s, entry_parent)
```

Comment:
Is this implicitly relying on the fact that `serialize_with_id` will return `false` for this parent? Or what's going on here?

It seems that `has_elem_basic_encoding(X)` has two meanings at once:
1. the element of `X` have a "basic encoding"
2. `X` itself can be encoded in a "basic" fashion, namely we don't need to encode any of its baserings etc. ?

I think it might help to separate these two concepts into two traits (even if in the end they will always have equal value)

Reply:
I've added a note above has_elem_basic_encoding definition, to clarify what is meant and how it is used